### ISSUE TITLE
Some issues with complex and single and double precision

### ISSUE THREAD

**Issue by cycomanic** [serge-sans-paille/pythran#1100]
I have some weird issues when I try to use single and double precision complex. The first issue is that if I try to compile a function for single and double precision alternatives like:

```python

 %%pythran 
    ...: import numpy
    ...: #pythran export test(complex128, float64)
    ...: #pythran export test(complex64, float32)
    ...: def test(x,y):
    ...:     return x.real*y + 1j*x.imag
    ...: 
```

I get an error:
```
PythranSyntaxError: Ambiguous overloads
        test(complex64, float32)
        test(complex128, float64).
```

Interestingly if I make either the two complex or the two floats an array. The compile works.

However if I implement the function using arrays it like so:
``` python
 %%pythran 
    ...: import numpy
    ...: #pythran export test(complex128[], float64[])
    ...: #pythran export test(complex64[], float32[])
    ...: def test(x,y):
    ...:     return x.real*y + 1j*x.imag

```
everything compiles however I get incorrect results:
```python
In [4]: x=np.array([2.3+1j]).astype(np.complex64)

In [5]: y=np.array([2.22]).astype(np.float32)

In [6]: test(x,y)
Out[6]: array([5.+1.j])

In [7]: x.real*y.real
Out[7]: array([5.106], dtype=float32)
```

**Comment by serge-sans-paille** [comment]
Thanks for the reports and detailed example. This troublesome, I'll have
a look at it :-/

**Comment by serge-sans-paille** [comment]
ok, those all are conversion issues, I'll handle that shortly

**Comment by serge-sans-paille** [comment]
Should be fixed by #1105 ! Thanks for reporting this issue, pythran is in much better state with that!

Can you just confirm everything is ok with it?

**Comment by cycomanic** [comment]
This has fixed the issue. I'm chasing down another issue that seems to be related, but I'm not sure if it was introduced by #1105. I'll try to make a test-case
